# HG changeset patch
# User Kai T. Ohlhus <k.ohlhus@gmail.com>
# Date 1545671398 -3600
#      Mon Dec 24 18:09:58 2018 +0100
# Branch stable
# Node ID 99318daeaddd7360afa54a4492123ae458ffa215
# Parent  ef7f95f89407140e3dffcc8eacd9c4036ea4e2fb
Properly deprecate output_max_field_width from cset 4d945f2e5914.

* libinterp/corefcn/pr-flt-fmt.cc (Foutput_max_field_width): Transform
deprecated function to m-file output_max_field_width.m

* scripts/deprecated/output_max_field_width.m: Create equivalent function.
* scripts/deprecated/module.mk: Add new m-file to build system.

* doc/interpreter/numbers.txi: Strip deprecated function from manual.
* NEWS: Advertise deprecated function.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -230,19 +230,19 @@ Summary of important user-visible change
       textread             | textscan
 
  ** Deprecated functions.
 
     The following functions have been deprecated in Octave 5.0 and will
     be removed from Octave 7 (or whatever version is the second major
     release after 5.0):
 
-      Function             | Replacement
-      ---------------------|------------------
-                           |
+      Function               | Replacement
+      -----------------------|------------------
+      output_max_field_width | output_precision
 
  ** The following functions were deprecated in Octave 4.2 and have been
     removed from Octave 5.0.
 
       Function             | Replacement
       ---------------------|------------------
       bitmax               | flintmax
       mahalanobis          | mahal in Octave Forge statistics pkg
diff --git a/doc/interpreter/numbers.txi b/doc/interpreter/numbers.txi
--- a/doc/interpreter/numbers.txi
+++ b/doc/interpreter/numbers.txi
@@ -285,24 +285,21 @@ may limit the maximum size of matrices t
 
 When you type a matrix or the name of a variable whose value is a
 matrix, Octave responds by printing the matrix in with neatly aligned
 rows and columns.  If the rows of the matrix are too large to fit on the
 screen, Octave splits the matrix and displays a header before each
 section to indicate which columns are being displayed.  You can use the
 following variables to control the format of the output.
 
-@DOCSTRING(output_max_field_width)
-
 @DOCSTRING(output_precision)
 
 It is possible to achieve a wide range of output styles by using
-different values of @code{output_precision} and
-@code{output_max_field_width}.  Reasonable combinations can be set using
-the @code{format} function.  @xref{Basic Input and Output}.
+different values of @code{output_precision}.  Reasonable combinations can be
+set using the @code{format} function.  @xref{Basic Input and Output}.
 
 @DOCSTRING(split_long_rows)
 
 Octave automatically switches to scientific notation when values become
 very large or very small.  This guarantees that you will see several
 significant figures for every value in a matrix.  If you would prefer to
 see all values in a matrix printed in a fixed point format, you can set
 the built-in variable @code{fixed_point_format} to a nonzero value.  But
diff --git a/libinterp/corefcn/pr-flt-fmt.cc b/libinterp/corefcn/pr-flt-fmt.cc
--- a/libinterp/corefcn/pr-flt-fmt.cc
+++ b/libinterp/corefcn/pr-flt-fmt.cc
@@ -40,26 +40,16 @@ output_precision (void)
 }
 
 void
 set_output_prec (int prec)
 {
   Voutput_precision = prec;
 }
 
-DEFUN (output_max_field_width, , ,
-       doc: /* -*- texinfo -*-
-@deftypefn {} {} output_max_field_width
-This function is obsolete and will be removed from a future version
-of Octave.
-@end deftypefn */)
-{
-  return octave_value (20);
-}
-
 DEFUN (output_precision, args, nargout,
        doc: /* -*- texinfo -*-
 @deftypefn  {} {@var{val} =} output_precision ()
 @deftypefnx {} {@var{old_val} =} output_precision (@var{new_val})
 @deftypefnx {} {} output_precision (@var{new_val}, "local")
 Query or set the internal variable that specifies the minimum number of
 significant figures to display for numeric output.
 
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -1,15 +1,16 @@
 FCN_FILE_DIRS += scripts/deprecated
 
 %canon_reldir%_FCN_FILES = \
   %reldir%/chop.m \
   %reldir%/comma.m \
   %reldir%/desktop.m \
   %reldir%/java2mat.m \
+  %reldir%/output_max_field_width.m \
   %reldir%/paren.m \
   %reldir%/semicolon.m \
   %reldir%/tmpnam.m \
   %reldir%/toascii.m
 
 %canon_reldir%dir = $(fcnfiledir)/deprecated
 
 %canon_reldir%_DATA = $(%canon_reldir%_FCN_FILES)
diff --git a/scripts/deprecated/output_max_field_width.m b/scripts/deprecated/output_max_field_width.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/output_max_field_width.m
@@ -0,0 +1,49 @@
+## Copyright (C) 2018 John W. Eaton
+##
+## This file is part of Octave.
+##
+## Octave is free software: you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <https://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn  {} {@var{val} =} output_max_field_width ()
+## @deftypefnx {} {@var{old_val} =} output_max_field_width (@var{new_val})
+## @deftypefnx {} {} output_max_field_width (@var{new_val}, "local")
+##
+## @code{output_max_field_width} is deprecated and will be removed in Octave
+## version 7.  Use @code{output_precision} instead.
+##
+## Query or set the internal variable that specifies the maximum width
+## of a numeric output field.
+##
+## When called from inside a function with the @qcode{"local"} option, the
+## variable is changed locally for the function and any subroutines it calls.
+## The original variable value is restored when exiting the function.
+## @seealso{format, fixed_point_format, output_precision}
+## @end deftypefn
+
+## FIXME: DEPRECATED: Remove in version 7.
+
+function retval = output_max_field_width (varargin)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "output_max_field_width is obsolete and will be removed from a future version of Octave, please use output_precision instead");
+  endif
+
+  retval = 20;
+
+endfunction
